Src/DependencyCollector/Shared.Tests/Implementation/DesktopDiagnosticSourceHttpProcessingTests.cs

```diff
@@ -0,0 +1,307 @@
+﻿#if NET45
+namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Globalization;
+    using System.Linq;
+    using System.Net;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Common;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.DependencyCollector.Implementation.Operation;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.TestFramework;
+    using Microsoft.ApplicationInsights.Web.TestFramework;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    [TestClass]
+    public class DesktopDiagnosticSourceHttpProcessingTests
+    {
+        #region Fields
+        private const string RandomAppIdEndpoint = "http://app.id.endpoint"; // appIdEndpoint - this really won't be used for tests because of the app id provider override.
+        private const int TimeAccuracyMilliseconds = 50;
+        private Uri testUrl = new Uri("http://www.microsoft.com/");
+        private int sleepTimeMsecBetweenBeginAndEnd = 100;
+        private TelemetryConfiguration configuration;
+        private List<ITelemetry> sendItems;
+        private DesktopDiagnosticSourceHttpProcessing httpProcessingFramework;
+        #endregion //Fields
+
+        #region TestInitialize
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            this.configuration = new TelemetryConfiguration();
+            this.sendItems = new List<ITelemetry>();
+            this.configuration.TelemetryChannel = new StubTelemetryChannel { OnSend = item => this.sendItems.Add(item) };
+            this.configuration.InstrumentationKey = Guid.NewGuid().ToString();
+            this.httpProcessingFramework = new DesktopDiagnosticSourceHttpProcessing(this.configuration, new ObjectInstanceBasedOperationHolder(), /*setCorrelationHeaders*/ true, new List<string>(), RandomAppIdEndpoint);
+            var correlationIdLookupHelper = new CorrelationIdLookupHelper((string ikey) =>
+            {
+                // Pretend App Id is the same as Ikey
+                var tcs = new TaskCompletionSource<string>();
+                tcs.SetResult(ikey);
+                return tcs.Task;
+            });
+            this.httpProcessingFramework.OverrideCorrelationIdLookupHelper(correlationIdLookupHelper);
+        }
+
+        [TestCleanup]
+        public void Cleanup()
+        {
+        }
+        #endregion //TestInitiliaze
+
+        /// <summary>
+        /// Validates that OnRequestSend and OnResponseReceive sends valid telemetry.
+        /// </summary>
+        [TestMethod]
+        public void RddTestHttpProcessingFrameworkUpdateTelemetryName()
```

Comment:
can you modify all the test names to reflect what they are testing? HttpProcessing vs DesktopDiagnosticsSourceHttpProcessing?